Client/Frontend/Browser/Playlist/Cache/PlaylistCacheLoader.swift

```diff
@@ -0,0 +1,788 @@
+            // If the second we receive the src, we reload the page by doing the below HTML,
+            // It may not have received all info necessary to play the item such as MetadataInfo
+            // For now it works 100% of the time and it is safe to do it. If we come across such a website, that causes problems,
+            // we'll need to find a different way of forcing the WebView to STOP loading metadata in the background
+            DispatchQueue.main.async {
+                self.webLoader?.tab.webView?.loadHTMLString("<html><body>PlayList</body></html>", baseURL: nil)
+            }
+        }
+    }
+    
+    private struct PageInfo: Codable {
+        let pageLoad: Bool
+        
+        public static func from(message: WKScriptMessage) -> PageInfo? {
+            if !JSONSerialization.isValidJSONObject(message.body) {
+                return nil
+            }
+            
+            do {
+                let data = try JSONSerialization.data(withJSONObject: message.body, options: [.fragmentsAllowed])
+                return try JSONDecoder().decode(PageInfo.self, from: data)
+            } catch {
+                log.error("Error Decoding PageInfo: \(error)")
+            }
+            
+            return nil
+        }
+    }
+}
+
+extension PlaylistWebLoader: WKNavigationDelegate {
+    // Recognize an Apple Maps URL. This will trigger the native app. But only if a search query is present. Otherwise
+    // it could just be a visit to a regular page on maps.apple.com.
+    fileprivate func isAppleMapsURL(_ url: URL) -> Bool {
+        if url.scheme == "http" || url.scheme == "https" {
+            if url.host == "maps.apple.com" && url.query != nil {
+                return true
+            }
+        }
+        return false
+    }
+
+    // Recognize a iTunes Store URL. These all trigger the native apps. Note that appstore.com and phobos.apple.com
+    // used to be in this list. I have removed them because they now redirect to itunes.apple.com. If we special case
+    // them then iOS will actually first open Safari, which then redirects to the app store. This works but it will
+    // leave a 'Back to Safari' button in the status bar, which we do not want.
+    fileprivate func isStoreURL(_ url: URL) -> Bool {
+        if url.scheme == "http" || url.scheme == "https" {
+            if url.host == "itunes.apple.com" {
+                return true
+            }
+        }
+        if url.scheme == "itms-appss" || url.scheme == "itmss" {
+            return true
+        }
+        return false
+    }
+
+    // This is the place where we decide what to do with a new navigation action. There are a number of special schemes
+    // and http(s) urls that need to be handled in a different way. All the logic for that is inside this delegate
+    // method.
+    
+    fileprivate func isUpholdOAuthAuthorization(_ url: URL) -> Bool {
```

Comment:
Where do we use this? Is `rewards` scheme supported by brave-ios?

Reply:
Fixed.